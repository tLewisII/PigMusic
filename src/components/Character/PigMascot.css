.pig-mascot {
  position: relative;
  width: 200px;
  height: 200px;
  cursor: pointer;
}

.pig-body {
  width: 150px;
  height: 120px;
  background: #FFB6C1;
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.1);
}

.pig-ears {
  position: absolute;
  top: -20px;
  width: 100%;
}

.ear {
  width: 40px;
  height: 50px;
  background: #FF91A4;
  position: absolute;
  transform-origin: bottom center;
}

.ear.left {
  left: 20px;
  transform: rotate(-30deg);
  border-radius: 50% 50% 0 70%;
}

.ear.right {
  right: 20px;
  transform: rotate(30deg);
  border-radius: 50% 50% 70% 0;
}

.pig-face {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.eyes {
  display: flex;
  gap: 40px;
  font-size: 24px;
  margin-bottom: 10px;
}

.eye {
  display: inline-block;
}

.snout {
  width: 60px;
  height: 40px;
  background: #FF91A4;
  border-radius: 50%;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 15px;
  margin-bottom: 10px;
}

.nostril {
  width: 10px;
  height: 14px;
  background: #D1707D;
  border-radius: 50%;
}

.mouth {
  font-size: 20px;
  font-weight: bold;
}


.pig-mascot.happy .pig-body {
  animation: wiggle 0.5s ease-in-out;
}

.pig-mascot.encouraging .snout {
  animation: snort 0.3s ease-in-out 2;
}

.pig-mascot.celebrating {
  animation: dance 1s ease-in-out infinite;
}

.celebration-effects {
  position: absolute;
  top: -40px;
  left: 50%;
  transform: translateX(-50%);
  display: flex;
  gap: 20px;
}

.confetti {
  font-size: 24px;
  animation: float 2s ease-in-out infinite;
}

.confetti:nth-child(2) {
  animation-delay: 0.3s;
}

.confetti:nth-child(3) {
  animation-delay: 0.6s;
}

@keyframes wiggle {
  0%, 100% { transform: translate(-50%, -50%) rotate(0deg); }
  25% { transform: translate(-50%, -50%) rotate(-5deg); }
  75% { transform: translate(-50%, -50%) rotate(5deg); }
}

@keyframes snort {
  0%, 100% { transform: scale(1); }
  50% { transform: scale(1.2); }
}

@keyframes dance {
  0%, 100% { transform: translateY(0) rotate(0deg); }
  25% { transform: translateY(-10px) rotate(-5deg); }
  75% { transform: translateY(-10px) rotate(5deg); }
}

@keyframes float {
  0%, 100% { transform: translateY(0); }
  50% { transform: translateY(-20px); }
}

/* Mobile responsive styles */
@media (max-width: 768px) {
  .pig-mascot {
    width: 100px;
    height: 100px;
  }

  .pig-body {
    width: 75px;
    height: 60px;
  }

  .pig-ears {
    top: -10px;
  }

  .ear {
    width: 20px;
    height: 25px;
  }

  .ear.left {
    left: 10px;
  }

  .ear.right {
    right: 10px;
  }

  .eyes {
    gap: 20px;
    font-size: 14px;
    margin-bottom: 5px;
  }

  .snout {
    width: 30px;
    height: 20px;
    gap: 8px;
    margin-bottom: 5px;
  }

  .nostril {
    width: 5px;
    height: 7px;
  }

  .mouth {
    font-size: 12px;
  }

  .celebration-effects {
    top: -20px;
    gap: 10px;
  }

  .confetti {
    font-size: 14px;
  }
}