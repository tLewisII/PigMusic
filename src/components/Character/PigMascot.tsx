import React from 'react';
import { useSpring, animated } from 'react-spring';
import { CustomizationItem } from '../../types/game';
import './PigMascot.css';

interface PigMascotProps {
  mood: 'idle' | 'happy' | 'encouraging' | 'celebrating';
  customization?: CustomizationItem[];
}

const PigMascot: React.FC<PigMascotProps> = ({ mood, customization = [] }) => {
  const bounceAnimation = useSpring({
    from: { transform: 'scale(1) translateY(0px)' },
    to: async (next) => {
      if (mood === 'happy' || mood === 'celebrating') {
        await next({ transform: 'scale(1.1) translateY(-20px)' });
        await next({ transform: 'scale(1) translateY(0px)' });
      }
    },
    config: { tension: 300, friction: 10 }
  });

  const wiggleAnimation = useSpring({
    from: { transform: 'rotate(0deg)' },
    to: async (next) => {
      if (mood === 'idle') {
        while (true) {
          await next({ transform: 'rotate(-3deg)' });
          await next({ transform: 'rotate(3deg)' });
          await next({ transform: 'rotate(0deg)' });
          await new Promise(resolve => setTimeout(resolve, 2000));
        }
      }
    },
    config: { tension: 200, friction: 15 }
  });

  const getMoodExpression = () => {
    switch (mood) {
      case 'happy':
        return { eyes: '^^', mouth: 'D' };
      case 'encouraging':
        return { eyes: 'oo', mouth: 'o' };
      case 'celebrating':
        return { eyes: '><', mouth: 'D' };
      default:
        return { eyes: 'â€¢â€¢', mouth: 'áµ•' };
    }
  };

  const expression = getMoodExpression();

  return (
    <animated.div 
      className={`pig-mascot ${mood}`}
      style={mood === 'idle' ? wiggleAnimation : bounceAnimation}
    >
      <div className="pig-body">
        <div className="pig-ears">
          <div className="ear left"></div>
          <div className="ear right"></div>
        </div>
        
        <div className="pig-face">
          <div className="eyes">
            <span className="eye left">{expression.eyes[0]}</span>
            <span className="eye right">{expression.eyes[1]}</span>
          </div>
          
          <div className="snout">
            <div className="nostril left"></div>
            <div className="nostril right"></div>
          </div>
          
          <div className="mouth">{expression.mouth}</div>
        </div>

        {customization.map(item => {
          if (item.type === 'hat') {
            return <div key={item.id} className={`accessory hat ${item.id}`}></div>;
          }
          return null;
        })}
      </div>

      {mood === 'celebrating' && (
        <div className="celebration-effects">
          <span className="confetti">ðŸŽ‰</span>
          <span className="confetti">âœ¨</span>
          <span className="confetti">ðŸŽŠ</span>
        </div>
      )}
    </animated.div>
  );
};

export default PigMascot;