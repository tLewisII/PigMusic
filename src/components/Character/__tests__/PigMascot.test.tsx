import React from 'react';
import { render } from '@testing-library/react';
import PigMascot from '../PigMascot';

describe('PigMascot', () => {
  it('renders correctly in idle mood', () => {
    const { container } = render(<PigMascot mood="idle" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly in happy mood', () => {
    const { container } = render(<PigMascot mood="happy" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly in encouraging mood', () => {
    const { container } = render(<PigMascot mood="encouraging" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly in celebrating mood', () => {
    const { container } = render(<PigMascot mood="celebrating" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('applies correct CSS classes for mood', () => {
    const { container } = render(<PigMascot mood="happy" />);
    const pigElement = container.querySelector('.pig-mascot');
    expect(pigElement).toHaveClass('pig-mascot', 'happy');
  });

  it('renders all pig body parts', () => {
    const { container } = render(<PigMascot mood="idle" />);
    expect(container.querySelector('.pig-body')).toBeInTheDocument();
    expect(container.querySelector('.pig-head')).toBeInTheDocument();
    expect(container.querySelector('.pig-snout')).toBeInTheDocument();
    expect(container.querySelectorAll('.nostril')).toHaveLength(2);
    expect(container.querySelectorAll('.eye')).toHaveLength(2);
    expect(container.querySelectorAll('.ear')).toHaveLength(2);
    expect(container.querySelectorAll('.leg')).toHaveLength(4);
    expect(container.querySelector('.tail')).toBeInTheDocument();
  });
});