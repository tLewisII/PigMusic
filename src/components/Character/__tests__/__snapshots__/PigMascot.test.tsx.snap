// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PigMascot renders correctly in celebrating mood 1`] = `
<div
  class="pig-mascot celebrating"
  style="transform: scale(1) translateY(0px);"
>
  <div
    class="pig-body"
  >
    <div
      class="pig-ears"
    >
      <div
        class="ear left"
      />
      <div
        class="ear right"
      />
    </div>
    <div
      class="pig-face"
    >
      <div
        class="eyes"
      >
        <span
          class="eye left"
        >
          &gt;
        </span>
        <span
          class="eye right"
        >
          &lt;
        </span>
      </div>
      <div
        class="snout"
      >
        <div
          class="nostril left"
        />
        <div
          class="nostril right"
        />
      </div>
      <div
        class="mouth"
      >
        D
      </div>
    </div>
  </div>
  <div
    class="celebration-effects"
  >
    <span
      class="confetti"
    >
      ðŸŽ‰
    </span>
    <span
      class="confetti"
    >
      âœ¨
    </span>
    <span
      class="confetti"
    >
      ðŸŽŠ
    </span>
  </div>
</div>
`;

exports[`PigMascot renders correctly in encouraging mood 1`] = `
<div
  class="pig-mascot encouraging"
  style="transform: scale(1) translateY(0px);"
>
  <div
    class="pig-body"
  >
    <div
      class="pig-ears"
    >
      <div
        class="ear left"
      />
      <div
        class="ear right"
      />
    </div>
    <div
      class="pig-face"
    >
      <div
        class="eyes"
      >
        <span
          class="eye left"
        >
          o
        </span>
        <span
          class="eye right"
        >
          o
        </span>
      </div>
      <div
        class="snout"
      >
        <div
          class="nostril left"
        />
        <div
          class="nostril right"
        />
      </div>
      <div
        class="mouth"
      >
        o
      </div>
    </div>
  </div>
</div>
`;

exports[`PigMascot renders correctly in happy mood 1`] = `
<div
  class="pig-mascot happy"
  style="transform: scale(1) translateY(0px);"
>
  <div
    class="pig-body"
  >
    <div
      class="pig-ears"
    >
      <div
        class="ear left"
      />
      <div
        class="ear right"
      />
    </div>
    <div
      class="pig-face"
    >
      <div
        class="eyes"
      >
        <span
          class="eye left"
        >
          ^
        </span>
        <span
          class="eye right"
        >
          ^
        </span>
      </div>
      <div
        class="snout"
      >
        <div
          class="nostril left"
        />
        <div
          class="nostril right"
        />
      </div>
      <div
        class="mouth"
      >
        D
      </div>
    </div>
  </div>
</div>
`;

exports[`PigMascot renders correctly in idle mood 1`] = `
<div
  class="pig-mascot idle"
  style="transform: rotate(0deg);"
>
  <div
    class="pig-body"
  >
    <div
      class="pig-ears"
    >
      <div
        class="ear left"
      />
      <div
        class="ear right"
      />
    </div>
    <div
      class="pig-face"
    >
      <div
        class="eyes"
      >
        <span
          class="eye left"
        >
          â€¢
        </span>
        <span
          class="eye right"
        >
          â€¢
        </span>
      </div>
      <div
        class="snout"
      >
        <div
          class="nostril left"
        />
        <div
          class="nostril right"
        />
      </div>
      <div
        class="mouth"
      >
        áµ•
      </div>
    </div>
  </div>
</div>
`;
